src/main/java/org/apache/parquet/hadoop/BdecParquetWriter.java

```diff
@@ -0,0 +1,285 @@
+import org.apache.parquet.column.ParquetProperties;
+import org.apache.parquet.column.values.factory.DefaultV1ValuesWriterFactory;
+import org.apache.parquet.crypto.FileEncryptionProperties;
+import org.apache.parquet.hadoop.api.WriteSupport;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+import org.apache.parquet.io.DelegatingPositionOutputStream;
+import org.apache.parquet.io.OutputFile;
+import org.apache.parquet.io.ParquetEncodingException;
+import org.apache.parquet.io.PositionOutputStream;
+import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.io.api.RecordConsumer;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.PrimitiveType;
+
+/**
+ * BDEC specific parquet writer.
+ *
+ * <p>Resides in parquet package because, it uses {@link InternalParquetRecordWriter} and {@link
+ * CodecFactory} that are package private.
+ */
+public class BdecParquetWriter implements AutoCloseable {
+  private final InternalParquetRecordWriter<List<Object>> writer;
+  private final CodecFactory codecFactory;
+
+  /**
+   * Creates a BDEC specific parquet writer.
+   *
+   * @param stream output
+   * @param schema row schema
+   * @param extraMetaData extra metadata
+   * @param channelName name of the channel that is using the writer
+   * @throws IOException
+   */
+  public BdecParquetWriter(
+      ByteArrayOutputStream stream,
+      MessageType schema,
+      Map<String, String> extraMetaData,
+      String channelName)
+      throws IOException {
+    OutputFile file = new ByteArrayOutputFile(stream);
+    ParquetProperties encodingProps = createParquetProperties();
+    Configuration conf = new Configuration();
+    WriteSupport<List<Object>> writeSupport =
+        new BdecWriteSupport(schema, extraMetaData, channelName);
+    WriteSupport.WriteContext writeContext = writeSupport.init(conf);
+
+    ParquetFileWriter fileWriter =
+        new ParquetFileWriter(
+            file,
+            schema,
+            ParquetFileWriter.Mode.CREATE,
+            ParquetWriter.DEFAULT_BLOCK_SIZE,
+            ParquetWriter.MAX_PADDING_SIZE_DEFAULT,
+            encodingProps.getColumnIndexTruncateLength(),
+            encodingProps.getStatisticsTruncateLength(),
+            encodingProps.getPageWriteChecksumEnabled(),
+            (FileEncryptionProperties) null);
+    fileWriter.start();
+
+    /* If our optimization for Parquet is enabled, there will be one writer per channel on each flush.
+    The memory will be allocated for each writer at the beginning even if we don't write anything with each writer,
+    which is the case when we enable the optimization.
+    Hence, to avoid huge memory allocations, we have to use CodecFactory
+    and move the BdecParquetWriter class in the parquet.hadoop package. */
+    codecFactory = new CodecFactory(conf, ParquetWriter.DEFAULT_PAGE_SIZE);
+    @SuppressWarnings("deprecation") // Parquet does not support the new one now
```

Comment:
could you use something supported instead of the deprecated one?

Reply:
unfortunately no, because InternalParquetRecordWriter doesn't support the new BytesInputCompressor.